bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/Resource.java

```diff
@@ -87,6 +86,11 @@ protected void finalize() {
 	 */
 	Device device;
 
+	/**
+	 * Invokes {@link ResourceTracker#run()} once the resource is eligible for GC
+	 */
+	private static final Cleaner cleaner = Cleaner.create();
```

Comment:
We can pass our own ThreadFactory to have a nicer thread name, but IMHO this adds too much complexity for what it's worth.

Reply:
FWIW it is really nice to have as many threads named when tracking down some problems. A while ago someone updated worker threads to have their name be that of the org.eclipse.core.runtime.jobs.Job running and that is super useful. 

I think you need to do this to give it a name:

```suggestion
	private static final Cleaner cleaner = Cleaner.create((r) -> new Thread(r, "SWTResourceTracker")); //$NON-NLS-1$
```

But the thread created by Cleaner.create() has a bunch of special properties (its an InnocuousThread) and it has higher than normal priority too.

Therefore, while I would love a name on it too, the default name of Cleaner-X (X is incrementing number) is probably great.

Reply:
I'll try to mimic the built-in factory. It's a little sad that the API doesn't provide an easy way to use custom thread names without reinventing the wheel :frowning_face:

Reply:
> Therefore, while I would love a name on it too, the default name of Cleaner-X (X is incrementing number) is probably great.

I also agree that in this case the default name might be enough.

That said, the following hack might work to set the thread name indirectly without a factory:

```
        cleaner.register(new Object(), new Runnable() {
            
            @Override
            public void run() {
                Thread.currentThread().setName("SWT-Cleaner-Thread");
            }
        }).clean();
```

Reply:
> That said, the following hack might work to set the thread name indirectly without a factory:

I don't think this will work, or rather not completely because it will set the name of the thread when `new Object()` is garbage collected, but then that thread will be done and future cleanables will run in a new thread.